### PR TITLE
add cf python client and check if org manager

### PR DESCRIPTION
**Pull request by dandersonsw**
## Changes proposed in this pull request:
- Adds CF Python Client
- Checks if user sending invite is an org manager, and if not it declines the request
-

## security considerations
Addition of the CF Python Client.
Improves security by limiting who can invite users.

### CODE REVIEW

File: uaaextras/clients/cf.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def get_cf_client(target_endpoint, access_token):
+    client = CloudFoundryClient(target_endpoint)
+    client._access_token = access_token
+    return client
+
+def is_org_manager(client, user_id):
+    if 'organization_manager' in client.v3.roles.list(user_guids=user_id):
+        return True
+    else:
+        return False
```
Comment (bengerman13) [inline uaaextras/clients/cf.py:16]: maybe just:
```
return 'organization_manager' in client.v3.roles.list(user_guids=user_id)
```

File: uaaextras/clients/cf.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    """
+
+    def __init__(self, target_endpoint, cf_user, cf_pass):
+        self.target_endpoint = target_endpoint
+        self.cf_user = cf_user
+        self.cf_pass = cf_pass
+
+    def _get_cf_client(self):
+        client = CloudFoundryClient(self.target_endpoint)
+        client.init_with_user_credentials(self.cf_user, self.cf_pass)
```
Comment (bengerman13) [inline uaaextras/clients/cf.py:19]: is it possible we could use the token the user gave us here, so we don't have to involve another set of credentials?

File: uaaextras/webapp.py [hunk 3]
```diff
@@ -506,6 +509,24 @@ def invite():
...
+            logging.exception("An invalid access token was decoded")
+            return render_template("error/token_validation.html"), 401
+        
+        user_id = decoded_token["user_id"]
+
+        #check for org_manager role
+        cf_client = CFClient(app.config["CF_API_URL"], app.config["CF_USER"], app.config["CF_PASS"])
+        if not cf_client.is_org_manager(cf_client._get_cf_client(), user_id):
+            logging.info('non-org-manager attempted user invite')
+            flash("You must be an org manager to invite a user.")
```
Comment (bengerman13) [inline uaaextras/webapp.py:527]: I think we could make this message friendlier, maybe:
```
Only organization managers can invite users. See our documentation here <link>
```

File: uaaextras/clients/cf.py (hunk 2 above)
Comment (bengerman13) [reply uaaextras/clients/cf.py:19]: it looks like yes: https://github.com/antechrestos/OAuth2Client/blob/master/main/oauth2_client/credentials_manager.py#L146

Comment (dandersonsw) [reply uaaextras/clients/cf.py]: Updated to use access token

File: uaaextras/webapp.py (hunk 3 above)
Comment (dandersonsw) [reply uaaextras/webapp.py]: Updated

File: uaaextras/clients/cf.py (hunk 1 above)
Comment (dandersonsw) [reply uaaextras/clients/cf.py]: Changed to properly grab organization manager

**Review by soutenniza — APPROVED** [review]
👍

**Pull request merged**